convert2rhel/subscription.py

```diff
@@ -630,19 +645,38 @@ def install_rhel_subscription_manager(pkgs_to_install, pkgs_to_upgrade=None):
 
     ..seealso:: :func:`_relevant_subscription_manager_pkgs` for the list of packages that we install.
     """
-
     pkgs_to_upgrade = pkgs_to_upgrade or []
     backedup_reposdir = backup.get_backedup_system_repos()
     varsdir = DEFAULT_YUM_VARS_DIR if system_info.version.major == 7 else DEFAULT_DNF_VARS_DIR
     backedup_varsdir = os.path.join(backup.BACKUP_DIR, hashlib.md5(varsdir.encode()).hexdigest())
 
+    setopts = []
+    # Oracle Linux 7 needs to set the obsoletes option to avoid installing the
+    # rhc-client-tools package instead of subscription-manager, as it is marked
+    # as obsolete in the subscription-manager specfile for OL7. This option in
+    # yum will make sure to ignore the obsoletes request and install the
+    # correct packages we are requesting.
+    # WARNING: This obsoletes will be applicable to all packages in the
+    # transaction.
+    if system_info.id == "oracle" and system_info.version.major == 7:
+        setopts.append("obsoletes=0")
+
+    client_tools_repofile = None
+    if system_info.id == "centos" and system_info.version.major == 8:
+        client_tools_repofile = repo.write_temporary_repofile(_CLIENT_TOOLS_RHEL_8_5_REPO)
+    else:
+        repofile_url = _CLIENT_TOOLS_REPOFILE_MAPPING[system_info.version.major]
+        client_tools_repofile = repo.download_repofile(repofile_url)
```

Comment:
The `download_repofile` function return None if unable to store the downloaded content. With that the code below would fail with a traceback. Better to change the download_repofile to raise a CriticalError instead of returning None.

Reply:
Okay, updated the code to raise exceptions in both `download_repofile` and `write_temporary_repofile`. Please re-review, @bocekm